converter/dicom-cast/src/Microsoft.Health.DicomCast.Core/Features/Worker/FhirTransaction/Patient/PatientPipelineStep.cs

```diff
@@ -71,9 +71,10 @@ public async Task PrepareRequestAsync(FhirTransactionContext context, Cancellati
             _patientSynchronizer.Synchronize(dataset, patient);
 
             if (requestMode == FhirTransactionRequestMode.None &&
-                !existingPatient.IsExactly(patient))
+                !existingPatient.IsEquivalent(patient))
```

Comment:
Can you explain what these logic supposed to do?

Reply:
This validates that the existing patient has the same identifiers (name and gender). This is so that a patient's birthdate is not taken into consideration when comparing between Dicom and Fhir (since we were concerned about inaccuracies and decided Fhir should be the source of truth).

Reply:
I don't think this works because if the name or gender changes, it wouldn't detect it as update right?

Reply:
Is this to make sure not to overwrite existing birthday? If that's the case, why not only set the birthday in `PatientBirthDateSynchronizer` if the Patient.Birthday is null?

Reply:
I think we don't even want to overwrite the birthdate if the patient existed before hand! And it should work since it still compares name and gender. If they are not deemed to be the same, it continues in here to set the mode to update.

Reply:
So what happens if a new field in Patient is being synced? Then the person has to known to also update this extension method to include that new property. The whole point of using the framework provided IsExactly is so that we don't have to have these logic manually written and let the system determine whether some property was updated or not.

Can you schedule some time with me to help understand what the code is trying to do?

Reply:
Sure! Let me set up time today.